pkg/security/ebpf/c/exec.h

```diff
@@ -527,7 +546,12 @@ int kprobe_do_exit(struct pt_regs *ctx) {
         struct proc_cache_t *pc = fill_process_context(&event.process);
         fill_container_context(pc, &event.container);
         fill_span_context(&event.span);
-        event.exit_code = (u32)PT_REGS_PARM1(ctx);
+        event.exit_code = (long)PT_REGS_PARM1(ctx);
+        u64 *in_coredump = (u64 *)bpf_map_lookup_elem(&tasks_in_coredump, &pid_tgid);
+        if (in_coredump) {
+            event.exit_code |= 0x80;
```

Comment:
We only need to switch this one bit ? We don't need to set an actual full value ?

Reply:
We first set the value using the `do_exit` function parameter, but then add this one bit in case it's missing for coredumps.